### ISSUE TITLE
[BUG] - OIDC with Authentik/Docker

### ISSUE THREAD

**Issue by CitizenStile** [mealie-recipes/mealie#4514]
### First Check

- [X] This is not a feature request.
- [X] I added a very descriptive title to this issue (title field is above this).
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Mealie documentation, with the integrated search.
- [X] I already read the docs and didn't find an answer.
- [ ] This issue can be replicated on the demo site (https://demo.mealie.io/).

### What is the issue you are experiencing?

When using OIDC with Authentik providing either a minimal OIDC config or full OIDC config via docker variables, mealie does not recognize that OIDC is configured nor does it offer to allow OIDC login through the configured server.

Along with trying an automatically generated Client ID and Secret, I have tried replacing them with simple values in Authentik and Mealie.

Notes:
* All domains replaced with selfhost.com for privacy
* Authentik/Mealie are on the same docker network
* OIDC Configuration URL matches between Authentik/Mealie
* Authentik is actively being used with Linkwarden/Paperless using OIDC

### Authentik Config
OAuth2 Provider
Name: Provider for Mealie
Authentication flow: default-authentication-flow (Welcome to authentik!)
Authorization flow: default-provider-authorization-implicit-consent (Authorize Application)

Client type: Confidential
Client ID: mealie
Client Secret: mealie123

Redirect URIs/Origins:
https://recipe.selfhost.com/login*
Signing Key: authentik Self-signed Certificate

### Docker Variables
* OIDC_AUTH_ENABLED:  true
* OIDC_CONFIGURATION_URL: https://auth.selfhost.com/application/o/mealie/.well-known/openid-configuration
* OIDC_CLIENT_ID: mealie
* OIDC_CLIENT_SECRET: mealie123
* Currently using postgres as the DB engine, but it was exactly the same using sqlite

### Steps to Reproduce

1) Add new container via Unraid
2) Repository: ghcr.io/mealie-recipes/mealie:v2.1.0
3) No Extra Parameters
4) No Post Arguments
5) Select docker network to align with Authentik
6) Add path /mnt/user/appdata/mealie:/app/data
7) Use provider settings from Authentik, creating a new OAuth2/OpenID provider if required. 
7a) Client Type set to Confidential
7b) Note Client ID/Client Secret for use in Mealie variables
7c) Set redirect URIs relative to Mealie domain
8) Startup mealie container, once loaded complete setup and check site settings for OIDC Readiness

### Please provide relevant logs

**Details**
Version: v2.1.0
Build: 53a566d08ab5b639671ee8abe1ad5a45dfdccc5a
Application Mode: Production
Demo Status: Not Demo
API Port: 9000
API Docs: Enabled
Database Type: postgres
Default Household: Family
Recipe Scraper Version: 15.2.1

**Checks**
Secure Site: Yes
Server Side Base URL: Yes
LDAP Ready: No
OIDC Ready: No
OpenAI Ready: No
Email Configured: Yes

-----

INFO     2024-11-04T21:10:07 - -----SYSTEM STARTUP-----
INFO     2024-11-04T21:10:07 - ------APP SETTINGS------
INFO     2024-11-04T21:10:07 - {
    "TESTING": false,
    "PRODUCTION": true,
    "LOG_CONFIG_OVERRIDE": null,
    "LOG_LEVEL": "debug",
    "theme": {
        "light_primary": "#E58325",
        "light_accent": "#007A99",
        "light_secondary": "#973542",
        "light_success": "#43A047",
        "light_info": "#1976D2",
        "light_warning": "#FF6D00",
        "light_error": "#EF5350",
        "dark_primary": "#E58325",
        "dark_accent": "#007A99",
        "dark_secondary": "#973542",
        "dark_success": "#43A047",
        "dark_info": "#1976D2",
        "dark_warning": "#FF6D00",
        "dark_error": "#EF5350"
    },
    "BASE_URL": "https://recipe.selfhosted.com",
    "STATIC_FILES": "/spa/static",
    "IS_DEMO": false,
    "HOST_IP": "*",
    "API_HOST": "0.0.0.0",
    "API_PORT": 9000,
    "API_DOCS": true,
    "TOKEN_TIME": 48,
    "GIT_COMMIT_HASH": "53a566d08ab5b639671ee8abe1ad5a45dfdccc5a",
    "ALLOW_SIGNUP": false,
    "DAILY_SCHEDULE_TIME": "23:45",
    "SECURITY_MAX_LOGIN_ATTEMPTS": 5,
    "SECURITY_USER_LOCKOUT_TIME": 24,
    "DB_ENGINE": "postgres",
    "DEFAULT_GROUP": "home",
    "DEFAULT_HOUSEHOLD": "Family",
    "SMTP_HOST": "smtp.gmail.com",
    "SMTP_PORT": "587",
    "SMTP_FROM_NAME": "Mealie",
    "SMTP_FROM_EMAIL": "changeme@gmail.com",
    "SMTP_AUTH_STRATEGY": "TLS",
    "LDAP_AUTH_ENABLED": false,
    "LDAP_SERVER_URL": null,
    "LDAP_TLS_INSECURE": false,
    "LDAP_TLS_CACERTFILE": null,
    "LDAP_ENABLE_STARTTLS": false,
    "LDAP_BASE_DN": null,
    "LDAP_QUERY_BIND": null,
    "LDAP_USER_FILTER": null,
    "LDAP_ADMIN_FILTER": null,
    "LDAP_ID_ATTRIBUTE": "uid",
    "LDAP_MAIL_ATTRIBUTE": "mail",
    "LDAP_NAME_ATTRIBUTE": "name",
    "OIDC_AUTH_ENABLED": true,
    "OIDC_CLIENT_ID": "mealie",
    "OIDC_CONFIGURATION_URL": "https://auth.selfhosted.com/application/o/mealie/.well-known/openid-configuration",
    "OIDC_SIGNUP_ENABLED": true,
    "OIDC_USER_GROUP": null,
    "OIDC_ADMIN_GROUP": null,
    "OIDC_AUTO_REDIRECT": false,
    "OIDC_PROVIDER_NAME": "OAuth",
    "OIDC_REMEMBER_ME": false,
    "OIDC_USER_CLAIM": "email",
    "OIDC_GROUPS_CLAIM": "groups",
    "OIDC_TLS_CACERTFILE": null,
    "OPENAI_BASE_URL": null,
    "OPENAI_MODEL": "gpt-4o",
    "OPENAI_CUSTOM_HEADERS": {},
    "OPENAI_CUSTOM_PARAMS": {},
    "OPENAI_ENABLE_IMAGE_SERVICES": true,
    "OPENAI_WORKERS": 2,
    "OPENAI_SEND_DATABASE_DATA": true,
    "OPENAI_REQUEST_TIMEOUT": 60,
    "WORKER_PER_CORE": 1,
    "UVICORN_WORKERS": 1
}

### Mealie Version

Mealie Version: v2.1.0
Build: 53a566d08ab5b639671ee8abe1ad5a45dfdccc5a

Authentik 2024.8.4

### Deployment

Unraid

**Comment by rafaelmathieu** [comment]
with version 2.0.0 and forward they updated the OIDC implementation.

Make sure to add "OIDC_CLIENT_SECRET" to your config.

In Authentik, you will have to update your mealie provider to the "confidential" client type. This will enable you to get the client secret that can be used for the "OIDC_CLIENT_SECRET" variable you just added to your docker config.

**Comment by CitizenStile** [comment]
Hi @rafaelmathieu, As I have documented the client secret is set and Authentik is set to confidential and the client ID and client Secret are matching between Authentik and Mealie

**Comment by rafaelmathieu** [comment]
I saw you said you did, but didn't see it in the config you posted. Maybe I missed it

"OIDC_AUTH_ENABLED": true,
"OIDC_CLIENT_ID": "mealie",
"OIDC_CONFIGURATION_URL": "https://auth.selfhosted.com/application/o/mealie/.well-known/openid-configuration",
"OIDC_SIGNUP_ENABLED": true,
"OIDC_USER_GROUP": null,
"OIDC_ADMIN_GROUP": null,
"OIDC_AUTO_REDIRECT": false,
"OIDC_PROVIDER_NAME": "OAuth",
"OIDC_REMEMBER_ME": false,
"OIDC_USER_CLAIM": "email",
"OIDC_GROUPS_CLAIM": "groups",
"OIDC_TLS_CACERTFILE": null,

**Comment by CitizenStile** [comment]
As far as I can gather, and from comments on the Authentik OAuth thread it does not show in the log when it prints out the variables and their values

**Comment by cmintey** [comment]
I have a PR (https://github.com/mealie-recipes/mealie/pull/4504) which should hopefully help to debug this issue

**Comment by CitizenStile** [comment]
> I have a PR (#4504) which should hopefully help to debug this issue

Swapped over to the nightly tag `ghcr.io/mealie-recipes/mealie:nightly` so that I could test out your patch. Turns out that while I did have the variable set, there was a trailing space. Seeing the reason that the various features are not working makes a huge difference, I hope that there is similar logic added in the UI as I doubt I'm the only person who runs into this kind of issue.

You sir, are a gentleman and a scholar! 

INFO     2024-11-08T20:43:03 - ------APP FEATURES------
INFO     2024-11-08T20:43:03 - --------==SMTP==--------
INFO     2024-11-08T20:43:03 - Enabled: True
INFO     2024-11-08T20:43:03 - --------==LDAP==--------
INFO     2024-11-08T20:43:03 - Enabled: False
Reason: LDAP_AUTH_ENABLED is false
INFO     2024-11-08T20:43:03 - --------==OIDC==--------
INFO     2024-11-08T20:43:03 - Enabled: False
Reason: Missing required values for ['OIDC_CLIENT_SECRET']
INFO     2024-11-08T20:43:03 - -------==OPENAI==-------
INFO     2024-11-08T20:43:03 - Enabled: False
Reason: OPENAI_API_KEY is not set
INFO     2024-11-08T20:43:03 - ------------------------
